kafka-rest/src/main/java/io/confluent/kafkarest/v2/KafkaConsumerManager.java

```diff
@@ -486,6 +494,87 @@ public ConsumerCommittedResponse committed(
     return response;
   }
 
+  /**
+   * Returns the beginning offset of the {@code topic} {@code partition}.
+   */
+  public long getBeginningOffset(String topic, int partition) {
+    log.debug("Beginning offset for topic {} and partition {}.", topic, partition);
+    KafkaConsumerState<?, ?, ?, ?> consumer;
+    try {
+      consumer =
+          retrying(
```

Comment:
can this really fail?
if it can (i don't think so, but good chance I'm missing something), i would question the need for this an make changes so it can't.
if that's justifiable for some reason, the retrying functionality would be better encapsulated lower down i think.